src/hiccup/compiler.clj

```diff
@@ -73,29 +73,53 @@
   (or content
       (and (html-mode?) (not (void-tags tag)))))
 
-(defn- merge-classes [class classes]
+(defn merge-classes [class classes]
   (cond
     (nil? class)    classes
     (string? class) (str classes " " class)
     :else           (str classes " " (str/join " " class))))
 
+(declare literal?)
+
+(defn- merge-classes-form [class-form classes]
+  (if (literal? class-form)
+    (merge-classes class-form classes)
+    `(merge-classes ~class-form ~classes)))
+
 (defn- merge-attributes [map-attrs id classes]
   (-> map-attrs
       (cond-> id      (assoc :id (or (:id map-attrs) id)))
       (cond-> classes (assoc :class (merge-classes (:class map-attrs) classes)))))
 
-(defn normalize-element
-  "Ensure an element vector is of the form [tag-name attrs content]."
-  [[tag & content]]
+(defn- merge-attributes-form [map-attrs id classes]
+  (-> map-attrs
+      (cond-> id      (assoc :id (or (:id map-attrs) id)))
+      (cond-> classes (assoc :class (merge-classes-form (:class map-attrs) classes)))))
+
+(defn- normalize-element*
+  "Ensure an element vector is of the form [tag-name attrs content. Merge the attributes
```

Comment:
Can you remove the docstring here? My view is that private functions shouldn't need them.